# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1334606318 14400
#      Mon Apr 16 15:58:38 2012 -0400
# Branch stable
# Node ID aa491bd9e19b56aad9548089fd8f8a2dc1a2be24
# Parent  05b7fa3b64c8a8b62c5fa08b491ff19c6f2cbb62
avoid unnecessary index overflow error when printing sparse array (bug #35148)

* ov-base-sparse.cc (octave_base_sparse<T>::print_info): Don't use
numel when computing percentage of non-zero elements.

diff --git a/src/ov-base-sparse.cc b/src/ov-base-sparse.cc
--- a/src/ov-base-sparse.cc
+++ b/src/ov-base-sparse.cc
@@ -313,17 +313,23 @@ octave_base_sparse<T>::print_raw (std::o
   // FIXME -- this should probably all be handled by a
   // separate octave_print_internal function that can handle format
   // compact, loose, etc.
 
   os << "Compressed Column Sparse (rows = " << nr
      << ", cols = " << nc
      << ", nnz = " << nz;
 
-  double dnel = matrix.numel ();
+  // Avoid calling numel here since it can easily overflow
+  // octave_idx_type even when there is no real problem storing the
+  // sparse array.
+
+  double dnr = nr;
+  double dnc = nc;
+  double dnel = dnr * dnc;
 
   if (dnel > 0)
     {
       double pct = (nz / dnel * 100);
 
       int prec = 2;
 
       // Display at least 2 significant figures and up to 4 as we
